### DISCUSSION TITLE
How do I default x86 & x64 and statics of everything by default?

### DISCUSSION THREAD

**Discussion by LedLoaf** [microsoft/vcpkg#19187] [Q&A]
Is there a way I can do this instead of typing 4 separate configurations for each library?

vcpkg install foo:x86-windows foo:x64-windows foo:x86-windows-static foo:x64-windows-static

Is there any way to just make this default the same way it always does x86 if you just type vcpkg install foo? Doing that for each library was annoying, because I had recently formatted at the time. 

I've searched around and it's very confusing.

Any help would be appreciated, I'm sure it's possible or a quicker way to type it. 

Thanks

**Comment by NancyLi1013** [comment]
Hi @LedLoaf 

Thanks for posting this issue.

Currently, there is no ways to support this. vcpkg will install x86-windows on Windows platform, x64-linux on Linux platform and x64-osx on macOS platform by default.

So you can use `vcpkg install foo` without typing the specific triplet on the above conditions.

But if you want to install the same port with different triplets by default, this is not supported in vcpkg now.

If you want to install different ports at the same time, you can refer to [this](https://github.com/microsoft/vcpkg/issues/15642#issuecomment-760691120).

**Comment by dg0yt** [comment]
You can also set the default triplet via an environment variable.
The tool also support "response files" which may collect any list of parameters, for reuse.
If you depend on a particular set of libraries, you might also create a single overlay port with those libraries as dependencies. When you build this port, it will trigger all builds as needed.

**Comment by LedLoaf** [comment]
Hey,

Thank you guys. Can any of you explain what a triplet is? I see it in the options ect and it seems to do with the different platforms?

So still a little bit confused after reading that thread. So just to clarify: vcpkg install foo --triplet=x64-windows ~~and that's going to get me the x86 and x64 and so I'm assuming I might have to do the same thing for static or are those acquired as well? Or does it mean anything therefore after will be x64 and if I want to switch to x86 I do it again?~~

Edit: So that didn't seem to do anything different? I just do this all on one line for example picked a random library:
vcpkg install byte-lite:x86-windows byte-lite:x64-windows byte-lite:x86-windows-static byte-lite:x64-windows-static

At the time used that as a template changing the words to a library and appending to the line. I'm not seeing the benefit and extra typing of --triplet.

@dg0yt I'm not exactly sure what I'm suppose to put in the environments? I just use those to make shortcut to paths. For the future I just backed up the whole vcpkg folder so I wouldn't have to do it again. But if something new came along, I thought it would be nice to cut down on typing x86/x64/x86-static/x64-static EACH time. 

Could anyone maybe explain a little bit to why this isn't a possible? It seems like something that would be added in already if it was possible. Perhaps, I can look up making a little script so I can just put in the name haha. That shouldn't be too hard.

**Comment by LedLoaf** [comment]
So, I figured a way to kind of do it. I've never programmed with PowerScript before but it works for me.

```
vcpkg_script.ps1

function Get-NewLib 
{
param([string]$lib)

$x86 = ":x86-windows"
$x64 = ":x64-windows"
$x86s = ":x86-windows-static"
$x64s = ":x64-windows-static"
$plat = "$lib$x86 $lib$x86s $lib$x64 $lib$x64s"

$Command = ("vcpkg install $plat")
Invoke-Expression -command ($Command)
}

In powershell:  Get-NewLib sfml 
```
Following the guide here to make a profile and put in the script so that you can run it at anytime.
https://docs.microsoft.com/en-us/previous-versions//bb613488(v=vs.85)?redirectedfrom=MSDN

Maybe that will help someone or someone could make the script better. Thanks

**Comment by NancyLi1013** [comment]
> Hey,
> 
> Thank you guys. Can any of you explain what a triplet is? I see it in the options ect and it seems to do with the different platforms?

I think [this ](https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md) doc can help you know what a triplet is.
